core/src/main/java/com/linecorp/armeria/common/stream/FixedStreamMessage.java

```diff
@@ -133,32 +136,68 @@ final void notifySubscriberOfCloseEvent(SubscriptionImpl subscription, CloseEven
 
     @Override
     final void cancel() {
-        cancelOrAbort(true);
+        cancelOrAbort(true, null);
     }
 
     @Override
     public final void abort() {
+        abort0(null);
+    }
+
+    @Override
+    public final void abort(Throwable cause) {
+        requireNonNull(cause, "cause");
+        abort0(() -> cause);
+    }
+
+    @Override
+    public final void abort(Supplier<? extends Throwable> causeSupplier) {
+        requireNonNull(causeSupplier, "causeSupplier");
+        abort0(causeSupplier);
+    }
+
+    private void abort0(@Nullable Supplier<? extends Throwable> causeSupplier) {
+        final Supplier<? extends Throwable> causeOrAbortStreamExceptionSupplier =
+                causeSupplier != null ? causeSupplier : AbortedStreamException::get;
+        final Throwable cause = requireNonNull(causeOrAbortStreamExceptionSupplier.get(),
```

Comment:
What is the point of using `Supplier` if you’re going to call `get()` regardless of whether `abort()` will fail?

Reply:
The reason why I call `get()` here is I thought there is no `abort()` fail in `FixedStreamMessage`.
If I missed something, please let me know 😊